src/store/dappChainStore.js

```diff
@@ -326,12 +317,22 @@ export default {
     async getMetamaskLoomBalance({ state , commit}, payload) {
       if (!state.web3) return 0
 
+      if (!state.dposUser) {
+        try {
+          await dispatch('initDposUser')
+        } catch (err) {
+          console.error("Error getting Loom balance", err)
+          return 0
+        }
+      }
       const web3js = state.web3
       const accounts = await web3js.eth.getAccounts()
       if (accounts.length === 0) return 0
       const address = accounts[0]
       try {
-        let balance = web3js.utils.fromWei(await state.LoomTokenInstance.methods
+        let loomWei = await state.dposUser.ethereumLoom.methods
```

Comment:
HELP Needed: Even though `dposUser` should be initialized based on the additions of L320-327, I am getting an error saying that `state.dposUser.ethereumLoom` is undefined, which was bizarre. Any ideas here? I really want to remove the `LoomTokenInstance` as it's useless outside of the `dposUser` container (its address is retrieved via the gateway contract)

Reply:
hmm, we do need the loom contract for listening to deposits fyi

Reply:
You are already logged in when that happens, so dposUser is already initialized.